ludwig/schema/preprocessing.py

```diff
@@ -187,28 +187,26 @@ class BinaryPreprocessingConfig(schema_utils.BaseMarshmallowConfig):
         description="What strategy to follow when there's a missing value in a binary column",
     )
 
-    fill_value: Union[int, float, str] = schema_utils.NumericOrStringOptionsField(
-        strings_utils.all_bool_strs(),
+    fill_value: Union[int, float, str] = schema_utils.OneOfOptionsField(
         default=None,
-        default_numeric=None,
-        default_option=None,
         allow_none=False,
-        min=0,
-        max=1,
         description="The value to replace missing values with in case the missing_value_strategy is fill_with_const",
+        field_options=[
+            schema_utils.FloatRange(default=None, allow_none=True, min=0, max=1),
```

Comment:
It's a good point as it definitely looks odd ("why are there 3 defaults, and why are two of them the same?"). 

The logic to do this would be a bit convoluted - the way we've constructed the other fields necessitates that `default` be specified explicitly or accepted implicitly with consequences for `allow_none` (e.g. if one of the options I want to add to `field_options` is a numeric field that should *disallow* `None`, then both `allow_none` and `default` have to be set accordingly or otherwise a validation error will occur). We could go back and decouple this relationship between the two parameters, but I might want to do that in a different PR, and I don't immediately know if it would be the right choice long-term either.

The stopgap in the meantime could be to accept `lambda`(s) as members of `field_options` rather than instances of marshmallow fields - and the lambda would basically set the `default` to the "supra"-default. That seems kind of messy too.

Reply:
Perhaps we could establish a convention that the first option in the list is the default validator?

Reply:
That's a possibility. Ultimately this is just a nit so if it's too much of a hassle to change for not that much value I would say lets just keep it as is.